selftest/selftest_test.go

```diff
@@ -67,3 +75,45 @@ func (s *selftestSuite) TestRunNotHappy(c *C) {
 	c.Check(err, IsNil)
 	c.Check(unhappyCheckRan, Equals, 1)
 }
+
+func (s *selftestSuite) TestUnexportedChecks(c *C) {
+
+	// collect what funcs we run in selftest.Check
+	var runCheckers []string
+	v := reflect.ValueOf(selftest.Checks)
+	for i := 0; i < v.Len(); i++ {
+		v := v.Index(i)
+		fname := runtime.FuncForPC(v.Pointer()).Name()
+		l := strings.Split(fname, ".")
+		runCheckers = append(runCheckers, l[len(l)-1])
+	}
+
+	// collect all "check*" functions
+	goFiles, err := filepath.Glob("*.go")
+	c.Assert(err, IsNil)
+	fset := token.NewFileSet()
+
+	var checkers []string
+	for _, fn := range goFiles {
+		f, err := parser.ParseFile(fset, fn, nil, 0)
+		c.Assert(err, IsNil)
+		ast.Inspect(f, func(n ast.Node) bool {
+			switch x := n.(type) {
+			case *ast.File:
+				return true
+			case *ast.FuncDecl:
+				name := x.Name.Name
```

Comment:
This is okay but this is funny to look at :)

Reply:
It is :) Upstream code!